src/resource.ts

```diff
@@ -0,0 +1,340 @@
+/* eslint-disable new-cap */
+import qs from 'qs';
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { cloneDeep } from 'lodash';
+
+import Model from './model';
+import List from './models/List';
+
+/**
+ * @deprecated since 2.2.0
+ */
+export type ResourceCallback = (error: any, resource: any) => void;
+
+export interface IParentParams {
+  resource: string;
+  id: string;
+}
+
+/**
+ * The base resource
+ */
+export default class Resource {
+  /**
+   * Resource code such as `payments`
+   * @var {string} resource
+   */
+  public static resource: string;
+  /**
+   * Refers to a Model class
+   * @var {Model} model
+   */
+  public static model: any;
+  /**
+   * @var {string} apiName
+   */
+  public readonly apiName: string;
+  /**
+   * @var {string} resourcePrefix
+   */
+  public readonly resourcePrefix: string;
+  /**
+   * @var {AxiosInstance} httpClient
+   */
+  protected readonly httpClient: AxiosInstance;
+  /**
+   * @var {string} parentId
+   */
+  protected parentId: string;
+
+  /**
+   * Constructor
+   */
+  public constructor(httpClient: AxiosInstance) {
+    this.httpClient = httpClient;
+  }
+
+  /**
+   * Error handler
+   */
+  protected static errorHandler(response: any, cb?: Function): any {
+    const error = (response && response.data) || response;
+
+    if (cb) {
+      return cb(error);
+    }
+    return error;
+  }
+
+  /**
+   * Set the parent ID by providing the parent
+   *
+   * @since 1.1.1
+   *
+   * @deprecated This method is not available
+   */
+  public withParent(parent: IParentParams): this {
+    if (parent && parent.id) {
+      this.setParentId(parent.id);
+    }
+    return this;
+  }
+
+  /**
+   * Create a resource by ID
+   *
+   * @params params - Resource-specific parameters
+   * @params cb - (DEPRECATED SINCE 2.2.0) Callback function, can be used instead of the returned Promise
+   *
+   * @returns The resource
+   *
+   * @since 1.0.0
+   *
+   * @public ✓ This method is part of the public API
+   */
+  public async create(prms: any, cb?: ResourceCallback): Promise<Model> {
+    const params = cloneDeep(prms);
+    const callback = typeof params === 'function' ? params : cb;
+    const query: any = {};
+    if (params !== null && typeof params === 'object') {
```

Comment:
Please use the `isPlainObject` method from `lodash`, as arrays are also objects:

```
typeof []
> "object"
```

Reply:
That one gets me every time :slightly_smiling_face: